src/lib/operators/join_hash.cpp

```diff
@@ -357,22 +357,35 @@ class JoinHash::JoinHashImpl : public AbstractReadOnlyOperatorImpl {
 
     Timer timer_materialization;
     if (_build_input_table->row_count() < _probe_input_table->row_count()) {
-      // When materializing the first side (here: the build side), we do not yet have a bloom filter. To keep the number
-      // of code paths low, materialize_*_side always expects a bloom filter. For the first step, we thus pass in a
-      // bloom filter that returns true for every probe.
+      // When materializing the first side (here: the build side), we do not yet have a Bloom filter. To keep the number
+      // of code paths low, materialize_*_side always expects a Bloom filter. For the first step, we thus pass in a
+      // Bloom filter that returns true for every probe.
       materialize_build_side(ALL_TRUE_BLOOM_FILTER);
       _performance.set_step_runtime(OperatorSteps::BuildSideMaterializing, timer_materialization.lap());
       materialize_probe_side(build_side_bloom_filter);
       _performance.set_step_runtime(OperatorSteps::ProbeSideMaterializing, timer_materialization.lap());
     } else {
+      std::cout << "probe larger build" << std::endl;
+      // Here, we first materialize the probe side and use the resulting Bloom filter in the materialization of the
+      // build side. Consequently, the passed Bloom filter in build() will have no effect as it has been already used
+      // during the materialization to filter non-matching values.
       materialize_probe_side(ALL_TRUE_BLOOM_FILTER);
       _performance.set_step_runtime(OperatorSteps::ProbeSideMaterializing, timer_materialization.lap());
       materialize_build_side(probe_side_bloom_filter);
       _performance.set_step_runtime(OperatorSteps::BuildSideMaterializing, timer_materialization.lap());
     }
 
+    // Store the number of materialized values. Depending on the order of materialization (which depends on the input
+    // sizes), each side might or might not be filtered by the Bloom filter.
+    for (const auto& partition : materialized_build_column) {
+      _performance.build_side_materialized_value_count += partition.elements.size();
```

Comment:
Slightly surprised about `_performance`. Didn't seem to have bugged me before. I wonder if `_performance_counters` might be easier to recognize. No strong opinion about it though.

Reply:
Changed to `_performance_data`, since it also stored bools etc.